packages/medusa/src/api-v2/admin/users/route.ts

```diff
@@ -43,16 +50,49 @@ export const POST = async (
   req: AuthenticatedMedusaRequest<CreateUserDTO>,
   res: MedusaResponse
 ) => {
-  const workflow = createUsersWorkflow(req.scope)
+  const userModuleService = req.scope.resolve<IUserModuleService>(
+    ModuleRegistrationName.USER
+  )
+  const authModuleService = req.scope.resolve<IAuthModuleService>(
+    ModuleRegistrationName.AUTH
+  )
+
+  const { jwt_secret } = req.scope.resolve("configModule").projectConfig
+
+  let authUser = await authModuleService.retrieve(req.auth.auth_user_id)
+  let token = jwt.sign(authUser, jwt_secret)
+
+  let user: UserDTO | null = null
+
+  if (req.auth.actor_id) {
+    user = await userModuleService.retrieve(req.auth.actor_id)
+
+    if (user.email !== req.validatedBody.email) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Request carries authentication for an existing user with a different email"
+      )
+    }
+
+    res.status(200).json({ user, token })
+
+    return
+  }
 
   const input = {
     input: {
-      users: [req.validatedBody],
+      userData: req.validatedBody,
+      authUserId: req.auth.auth_user_id,
     },
   }
 
-  const { result } = await workflow.run(input)
+  const { result } = await createUserAccountWorkflow(req.scope).run(input)
+
+  user = result
+
+  // Refresh token with updated auth user app_metadata
+  authUser = await authModuleService.retrieve(req.auth.auth_user_id)
+  token = jwt.sign(authUser, jwt_secret)
 
-  const [user] = result
-  res.status(200).json({ user })
+  res.status(200).json({ user, token })
```

Comment:
We respond with the user and token so the subsequent session token conversion is a simpler flow. This is purely convenience, so I am happy to remove it in favor of just the user if we think that makes for a better API.